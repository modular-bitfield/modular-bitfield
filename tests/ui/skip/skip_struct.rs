use modular_bitfield::prelude::*;

#[bitfield(skip(new, new))]
struct A {
    f: u8,
}

#[bitfield(skip(from_bytes, new, from_bytes))]
struct B {
    f: u8,
}

#[bitfield(skip(new, into_bytes, into_bytes))]
struct C {
    f: u8,
}

#[bitfield(skip(invalid))]
struct D {
    f: u8,
}

#[bitfield(skip)]
struct E {
    f: u8,
}

#[bitfield(skip(all, convert))]
struct F {
    f: u8,
}

#[bitfield(skip(all, new))]
struct G {
    f: u8,
}

#[bitfield(skip(convert, from_bytes, into_bytes))]
struct H {
    f: u8,
}

#[bitfield]
struct I {
    _implicit_skip: u8,
}

fn main() {
    let i = I::new();
    i.implicit_skip();
    i._implicit_skip();
}
