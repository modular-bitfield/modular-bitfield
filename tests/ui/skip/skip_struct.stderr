error: encountered duplicate `#[skip(new)]`
 --> tests/ui/skip/skip_struct.rs:3:22
  |
3 | #[bitfield(skip(new, new))]
  |                      ^^^

error: duplicate `#[skip(new)]` here
 --> tests/ui/skip/skip_struct.rs:3:17
  |
3 | #[bitfield(skip(new, new))]
  |                 ^^^

error: encountered duplicate `#[skip(from_bytes)]`
 --> tests/ui/skip/skip_struct.rs:8:34
  |
8 | #[bitfield(skip(from_bytes, new, from_bytes))]
  |                                  ^^^^^^^^^^

error: duplicate `#[skip(from_bytes)]` here
 --> tests/ui/skip/skip_struct.rs:8:17
  |
8 | #[bitfield(skip(from_bytes, new, from_bytes))]
  |                 ^^^^^^^^^^

error: encountered duplicate `#[skip(into_bytes)]`
  --> tests/ui/skip/skip_struct.rs:13:34
   |
13 | #[bitfield(skip(new, into_bytes, into_bytes))]
   |                                  ^^^^^^^^^^

error: duplicate `#[skip(into_bytes)]` here
  --> tests/ui/skip/skip_struct.rs:13:22
   |
13 | #[bitfield(skip(new, into_bytes, into_bytes))]
   |                      ^^^^^^^^^^

error: encountered unknown or unsupported #[skip(..)] specifier
  --> tests/ui/skip/skip_struct.rs:18:17
   |
18 | #[bitfield(skip(invalid))]
   |                 ^^^^^^^

error: expected attribute arguments in parentheses: `skip(...)`
  --> tests/ui/skip/skip_struct.rs:23:12
   |
23 | #[bitfield(skip)]
   |            ^^^^

error: encountered duplicate `#[skip(from_bytes)]`
  --> tests/ui/skip/skip_struct.rs:28:22
   |
28 | #[bitfield(skip(all, convert))]
   |                      ^^^^^^^

error: duplicate `#[skip(from_bytes)]` here
  --> tests/ui/skip/skip_struct.rs:28:17
   |
28 | #[bitfield(skip(all, convert))]
   |                 ^^^

error: encountered duplicate `#[skip(new)]`
  --> tests/ui/skip/skip_struct.rs:33:22
   |
33 | #[bitfield(skip(all, new))]
   |                      ^^^

error: duplicate `#[skip(new)]` here
  --> tests/ui/skip/skip_struct.rs:33:17
   |
33 | #[bitfield(skip(all, new))]
   |                 ^^^

error: encountered duplicate `#[skip(from_bytes)]`
  --> tests/ui/skip/skip_struct.rs:38:26
   |
38 | #[bitfield(skip(convert, from_bytes, into_bytes))]
   |                          ^^^^^^^^^^

error: duplicate `#[skip(from_bytes)]` here
  --> tests/ui/skip/skip_struct.rs:38:17
   |
38 | #[bitfield(skip(convert, from_bytes, into_bytes))]
   |                 ^^^^^^^

error[E0599]: no method named `implicit_skip` found for struct `I` in the current scope
  --> tests/ui/skip/skip_struct.rs:50:7
   |
44 | struct I {
   | ------ method `implicit_skip` not found for this struct
...
50 |     i.implicit_skip();
   |       ^^^^^^^^^^^^^ method not found in `I`

error[E0599]: no method named `_implicit_skip` found for struct `I` in the current scope
  --> tests/ui/skip/skip_struct.rs:51:7
   |
44 | struct I {
   | ------ method `_implicit_skip` not found for this struct
...
51 |     i._implicit_skip();
   |       ^^^^^^^^^^^^^^ method not found in `I`
